### ISSUE TITLE
inputValue is not changing search result in async select

### ISSUE THREAD

**Issue by donnikitos** [JedWatson/react-select#4264]
**Case 1**
After dynamically changing _inputValue_ prop the search does not update.

**Case 2**
After initially declaring _initValue_ via state and then removing the _initValue_ via _onInputChange_ seems to keeps _initValue_ as it was declared initially.

**Live demo** of both cases with steps to reproduce
https://codesandbox.io/s/busy-sutherland-xx8o2?file=/example.js:1411-1424
**Product version**
v3.1.0(react-select)
**Affected browser(s)**
?

**Comment by ebonow** [comment]
Greetings @donnikitos ,

Just to verify, it seems you are trying to use one input to set inputValue for the Select and as it changes, allow it to both act as the setter for the Select inputValue as well as trigger the onChangeInput handler. Is this correct?

Assuming that is the case, I was able to replicate and verify the behavior you described. I believe the answer though is likely more related to native behavior as the the `onInputChange` is applied directly to the `onChange` handler of the input. 

As with a standard input, if the value prop is directly changed, the `onChange` will not be triggered. This means that the input itself needs to raise an onChange event to trigger the `onInputChange` method which in turn interacts with the async lookup.

Let me know if this makes sense. In regards to a solution for what you are working on, I began a sandbox here, but haven't completely resolved it yet. Hopefully this makes for a good start but the goal should be to have the first input onChange trigger the second.
https://codesandbox.io/s/react-select-controlled-async-input-7q7xy?file=/example.js
